test/AngularSpec.js

```diff
@@ -78,6 +78,42 @@ describe('angular', function() {
       expect(copy(objWithRegExp.re) === objWithRegExp.re).toBeFalsy();
     });
 
+    it("should copy a TypedArray with no destination", function() {
+      if (typeof Float32Array !== 'undefined') {
+        var arraysToCheck = [
+          Uint8Array, Uint16Array, Uint32Array, Int8Array, Int16Array,
+          Int32Array, Float32Array, Float64Array
+        ];
+
+        for (var i = 0; i < arraysToCheck.length; i++) {
+          var TypeOfArray = arraysToCheck[i];
+          // only run this test on browsers that support typed arrays
+          var src = new TypeOfArray(2);
+          src[1] = 1;
+          var dst = copy(src);
+          expect(copy(src) instanceof TypeOfArray).toBeTruthy();
+          expect(dst).toEqual(src);
+          expect(dst).not.toBe(src);
+        }
+      }
+    });
+
+    it("should throw an exception if a TypedArray is the destination", function() {
+      if (typeof Float32Array !== 'undefined') {
+        var arraysToCheck = [
+          Uint8Array, Uint16Array, Uint32Array, Int8Array, Int16Array,
```

Comment:
we should actually split these into multiple `it()` blocks... @matsko has a tool for doing this in https://github.com/angular/angular.js/pull/10864 but it has not landed yet. Matias do you think you could get that in soon or else submit a followup PR to cleanup these tests if that lands first?